frontend/src/components/patient/HistoryModal.jsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useState } from "react";
+import { useEffect } from "react";
+import formatFullDate from "../../utils/formatFullDate";
+
+import { fetchHistoryById } from "../../api/medicalRecordFunctions";
+
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Button,
+  useDisclosure,
+  Chip,
+} from "@heroui/react";
+
+export default function HistoryModal({
+  isOpen,
+  onOpenChange,
+  selectedHistory,
+}) {
+  const [selectedHistoryBody, setSelectedHistoryBody] = useState();
+
+  useEffect(() => {
+    async function fetchHistory() {
+      try {
+        const fetchedHistory = await fetchHistoryById(selectedHistory);
+        setSelectedHistoryBody(fetchedHistory);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+
+    fetchHistory();
+  }, [selectedHistory]);
+
+  return (
+    <Modal isOpen={isOpen} onOpenChange={onOpenChange}>
+      <ModalContent>
+        {(onClose) => (
+          <>
+            <ModalHeader className="flex flex-col gap-1 text-ca-black">
+              <h3>
+                {selectedHistoryBody
+                  ? formatFullDate(selectedHistoryBody.createdAt)
+                  : "History Body"}
+              </h3>
+            </ModalHeader>
+            <ModalBody className="max-h-96 overflow-scroll text-ca-black">
+              {selectedHistoryBody ? (
+                <p className="text-sm">
+                  <span className="font-bold">Doctor:</span>{" "}
+                  {selectedHistoryBody.doctor.name +
+                    " " +
+                    selectedHistoryBody.doctor.lastname}
+                </p>
+              ) : (
+                <></>
+              )}
+              {selectedHistoryBody ? (
+                <p className="text-sm">
+                  <span className="font-bold">Diagnosis:</span>{" "}
+                  {selectedHistoryBody.diagnosis}
+                </p>
+              ) : (
+                <></>
+              )}
+              {selectedHistoryBody ? (
+                <p className="text-sm">
+                  <span className="font-bold">Notes:</span>{" "}
+                  {selectedHistoryBody.notes}
+                </p>
```

Comment:
Seems `selectedHistoryBody` has been checked multiple times here, it should be possible to just combine them for better readability?